src/lib.rs

```diff
@@ -140,11 +178,67 @@ impl<'u> M<'u> {
     pub const fn up(sql: &'u str) -> Self {
         Self {
             up: sql,
+            up_hook: None,
             down: None,
+            down_hook: None,
             foreign_key_check: false,
         }
     }
 
+    /// Create a schema update running additional Rust code. The SQL command will be executed only
+    /// when the migration has not been executed on the underlying database. The `hook` code will
+    /// be executed *after* the SQL command executed successfully.
+    ///
+    /// See [`Self::up()`] for additional notes.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use rusqlite_migration::{M, Migrations};
+    /// use rusqlite::Transaction;
+    ///
+    /// let migrations = Migrations::new(vec![
+    ///     M::up_with_hook(
+    ///         "CREATE TABLE novels (text TEXT);",
+    ///         move |tx: &Transaction| {
+    ///             tx.execute("INSERT INTO novels (text) VALUES (?1)", ("Lorem Ipsum …",))
+    ///                 .unwrap();
+    ///
+    ///             Ok(())
+    ///         },
+    ///     ),
```

Comment:
This looks like the type of things you wouldn’t want to do in migrations (namely populating the table with user data). In the interest of keeping the documentation’s example closer to ideal usage, what do you think of just having:
```suggestion
    ///     // This table will later be filled with some novel content
    ///     M::up("CREATE TABLE novels (text TEXT);"),
    ///
```
?